### PR TITLE
Doc updates related to Operator v5.0.0

### PR DESCRIPTION
**Pull request by djwfyi**
Updates docs for most changes through v5.0.4.
Does not have changes related to sidecar containers.

Partially addresses #791

Staged: http://192.241.195.202:9000/staging/operator-5.0.0/index.html

### CODE REVIEW

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst [hunk 1]
```diff
@@ -36,15 +39,147 @@ The following table lists the upgrade paths from previous versions of the MinIO
...
+   - Your local host has ``kubectl`` installed and configured with access to the Kubernetes cluster
+
+This procedure upgrades the MinIO Operator from any 4.5.8 or later release to |operator-version-stable|.
+
+Breaking changes
+~~~~~~~~~~~~~~~~
+
+The following changes apply for Operator v5.0.0 or later:
+
+- The Operator's Graphical User Interface now comes bundled with the same container as the Operator.
```
Comment (feorlen) [inline source/operations/install-deploy-manage/upgrade-minio-operator.rst:61]: Does this mean previously they were in separate containers and now they are both in one? `bundled with the same container` suggested multiple containers to me, but I don't know if that's correct.

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst [hunk 2]
```diff
@@ -36,15 +39,147 @@ The following table lists the upgrade paths from previous versions of the MinIO
...
+- The ``.spec.s3`` field is replaced by the ``.spec.features`` field.
+- The ``.spec.credsSecret`` field is replaced by the ``.spec.configuration`` field.
+
+  This secret should hold all the environment variables for the MinIO deployment that contain sensitive information and should not show in ``.spec.env``.
```
Comment (feorlen) [inline source/operations/install-deploy-manage/upgrade-minio-operator.rst:65]: Is `this secret` referring to `.spec.credsSecret`? Perhaps something like `.spec.credsSecret contains sensitive environment variables that should not be shown in .spec.env."

File: source/reference/kubectl-minio-plugin.rst [hunk 3]
```diff
@@ -89,6 +89,71 @@ Subcommands
...
+The :ref:`MinIO Operator <minio-operator-installation>` uses the following environment variables during startup to set configuration settings.
+
+
+.. envvar:: MINIO_OPERATOR_CERTIFICATES_VERSION
+
+   Specifies the certificate API version to use.
+
+   Valid values are ``v1`` or ``v1beta1``.
+
+   When not specified, the default is the API the Kubernetes provides.
```
Comment (feorlen) [inline source/reference/kubectl-minio-plugin.rst:104]: > the API the Kubernetes provides
I suspect the second `the` is a typo

File: source/reference/kubectl-minio-plugin.rst [hunk 4]
```diff
@@ -89,6 +89,71 @@ Subcommands
...
+
+.. envvar:: MINIO_OPERATOR_RUNTIME
+
+   Specify the type of runtime to use.
+
+   Valid values are ``EKS``, ``Rancher``, or ``OpenShift``.
+   Leave blank if none of the value options apply.
```
Comment (feorlen) [inline source/reference/kubectl-minio-plugin.rst:111]: I'm not sure what `value options` are, perhaps just `options`?

File: source/reference/kubectl-minio-plugin.rst [hunk 5]
```diff
@@ -89,6 +89,71 @@ Subcommands
...
+
+   Override the default signer for certificate signing requests (CSRs).
+
+   When not specified, the default value is ``kubernetes.io/kubelet-serving``.
+
+.. envvar:: OPERATOR_CERT_PASSWD
+   
+   *Optional*
+
+   The password Operator should use to use to decrypt the private key in the TLS certificate for Operator.
```
Comment (feorlen) [inline source/reference/kubectl-minio-plugin.rst:125]: I suspect this was intended to be `The password Operator should use to decrypt...`. Perhaps an alternative is "The password Operator should use to decrypt the private key in the Operator TLS certificate" (although I'm not entirely clear the context).

**Review by feorlen — CHANGES_REQUESTED** [review]
Noted a couple typos, otherwise looks good.

The other comments are probably me not understanding. Use or ignore as you see fit.

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst [hunk 6]
```diff
@@ -36,15 +39,147 @@ The following table lists the upgrade paths from previous versions of the MinIO
...
+- Both the **Log Search API** (``.spec.log``) and **Prometheus** (``.spec.prometheus``) deployments have been removed.
+  However, existing deployments are left running as standalone deployments / statefulsets with no connection to the Tenant CR.
+  If the Tenant CR is deleted, this does not cascade to the log or Prometheus deployments.
+
+  .. important::
+
+     MinIO recommends that you create a yaml file to manage these deployments going forward.
+
+.. container:: procedure
```
Comment (ravindk89) [inline source/operations/install-deploy-manage/upgrade-minio-operator.rst:74]: Are we doing CSS here on this?

Maybe break this up into

H2. Tenant Custom Resource Definition Changes

H2. Log Search and Prometheus Removal

H2. Upgrade Procedure

So people can handle each thing in order.

File: source/reference/kubectl-minio-plugin.rst [hunk 7]
```diff
@@ -89,6 +89,71 @@ Subcommands
 - :mc:`~kubectl minio delete`
 - :mc:`~kubectl minio version`
 
+Environment Variables
+---------------------
+
+The :ref:`MinIO Operator <minio-operator-installation>` uses the following environment variables during startup to set configuration settings.
```
Comment (ravindk89) [inline source/reference/kubectl-minio-plugin.rst:95]: I assume these have to be set on the host where `kubectl init` runs, not on like the K8s hosts.

**Review by ravindk89 — CHANGES_REQUESTED** [review]
I think we need to tweak the pre-procedure

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst (hunk 1 above)
Comment (djwfyi) [reply source/operations/install-deploy-manage/upgrade-minio-operator.rst:61]: Removed this line.
For your knowledge, @feorlen, the GUI used to be part of the Console repo. It was moved in this release to the Operator repo.

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst (hunk 2 above)
Comment (djwfyi) [reply source/operations/install-deploy-manage/upgrade-minio-operator.rst:65]: I went ahead and swapped "This secret" for the actual field name instead. Also added your suggestion, @ravindk89.

File: source/reference/kubectl-minio-plugin.rst (hunk 4 above)
Comment (djwfyi) [reply source/reference/kubectl-minio-plugin.rst:111]: Dropped "value" as suggested. Thanks!

File: source/reference/kubectl-minio-plugin.rst (hunk 5 above)
Comment (djwfyi) [reply source/reference/kubectl-minio-plugin.rst:125]: Sometimes my fingers and mind don't quite stay in sync and one wants to do what the other already did.

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst (hunk 6 above)
Comment (djwfyi) [reply source/operations/install-deploy-manage/upgrade-minio-operator.rst:74]: Not sure. I was just following what you had done last time you updated this page, which added this `.. procedure::` for each one. I can drop it.

Comment (ravindk89) [reply source/operations/install-deploy-manage/upgrade-minio-operator.rst:74]: I can't even begin to remember why I did that. It must be an scss thing, but I think at this point clever things can go away for things that work.

File: source/operations/install-deploy-manage/minio-operator-console.rst [hunk 8]
```diff
@@ -33,31 +33,35 @@ Connect to the Operator Console
...
    :alt: MinIO Operator Console
 
-You can :ref:`deploy a MinIO Tenant <minio-k8s-deploy-minio-tenant>` through the 
-Operator Console.
+You can :ref:`deploy a MinIO Tenant <minio-k8s-deploy-minio-tenant>` through the Operator Console.
 
-The Operator Console automatically detects any MinIO Tenants 
-deployed on the cluster, whether provisioned through the Operator Console 
-or through the :ref:`MinIO Kubernetes Plugin <minio-k8s-deploy-minio-tenant-commandline>`.
+The Operator Console automatically detects any MinIO Tenants deployed on the cluster, whether provisioned through the Operator Console or through the :ref:`MinIO Kubernetes Plugin <minio-k8s-deploy-minio-tenant-commandline>`.
```
[the remark this answers is not in the document]
Comment (djwfyi) [reply source/operations/install-deploy-manage/minio-operator-console.rst:46]: Good point. Reworked.

File: source/operations/install-deploy-manage/minio-operator-console.rst [hunk 9]
```diff
@@ -33,31 +33,35 @@ Connect to the Operator Console
...
 
+.. versionadded:: Operator 5.0.0
+
+   You can download a Log Report for a tenant from the Pods summary screen.
+
+   The report downloads as ``<tenant-name>-report.zip``.
+   The ZIP archive contains status, events, and log information for each pool on the deployment.
+   The archive also includes a summary yaml file describing the deployment.
+
+   If subscribed, you can upload this file to :ref:`SUBNET <minio-docs-subnet>` for analysis by MinIO Engineers.
```
[the remark this answers is not in the document]
Comment (djwfyi) [reply source/operations/install-deploy-manage/minio-operator-console.rst:59]: 👍

File: source/operations/install-deploy-manage/minio-operator-console.rst [hunk 10]
```diff
@@ -33,31 +33,35 @@ Connect to the Operator Console
...
+
 Tenant Registration
 -------------------
 
-.. versionchanged:: 0.19.5
-
-   You can register your MinIO tenants with your |SUBNET| account using the Operator Console.
+You can register your MinIO tenants with your |SUBNET| account using the Operator Console.
```
[the remark this answers is not in the document]
Comment (djwfyi) [reply source/operations/install-deploy-manage/minio-operator-console.rst:64]: Reworded.

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst [hunk 11]
```diff
@@ -36,15 +39,147 @@ The following table lists the upgrade paths from previous versions of the MinIO
...
+
+   1. Back up deployments
```
[the remark this answers is not in the document]
Comment (djwfyi) [reply source/operations/install-deploy-manage/upgrade-minio-operator.rst:76]: Changed this heading, and also reworked the presentation on the rest of it as the below comment suggested.

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst (hunk 6 above)
Comment (djwfyi) [reply source/operations/install-deploy-manage/upgrade-minio-operator.rst:74]: Removed the .. procedure throughout and added headings.

File: source/reference/kubectl-minio-plugin.rst (hunk 7 above)
Comment (djwfyi) [reply source/reference/kubectl-minio-plugin.rst:95]: Added a note to that effect.

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst [hunk 12]
```diff
@@ -36,15 +39,147 @@ The following table lists the upgrade paths from previous versions of the MinIO
...
+         kubectl -n $NAMESPACE get svc $TENANT_NAME-log-hl-svc -o yaml > $TENANT_NAME-log-hl-svc.yaml
+         kubectl -n $NAMESPACE get svc $TENANT_NAME-log-search-api -o yaml > $TENANT_NAME-log-search-api.yaml
+         kubectl -n $NAMESPACE get svc $TENANT_NAME-prometheus-hl-svc -o yaml > $TENANT_NAME-prometheus-hl-svc.yaml
+
+      - Replace ``myminio`` with the name of the tenant you are upgrading.
+      - Replace ``mynamespace`` with the namespace for the tenant you are upgrading.
+
+   2. Remove ``.metadata.ownerReferences`` for all backed up files
+
+   3. *(Optional)* To continue using Log Search API and Prometheus, add the following variables to ``.spec.env``
```
[the remark this answers is not in the document]
Comment (djwfyi) [reply source/operations/install-deploy-manage/upgrade-minio-operator.rst:97]: Clarified with @cniackz and updated this step with better instructions.

**Review by feorlen — APPROVED** [review]
LGTM

**Pull request merged**